clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -306,6 +306,36 @@ static int64_t getIntExprValue(const Expr *E, ASTContext &Ctx) {
   return E->getIntegerConstantExpr(Ctx)->getSExtValue();
 }
 
+// Collect function attributes related to SYCL
+static void collectSYCLAttributes(Sema &S, FunctionDecl *FD,
```

Comment:
`const FunctionDecl *` ?

Reply:
Adding a const here leads to a compilation error on line 334, because the call FD->dropAttr(... ) does not like a const "this" operand. dropAttr is an existing function, outside the scope of this PR.

Reply:
Agreed, thanks for checking!